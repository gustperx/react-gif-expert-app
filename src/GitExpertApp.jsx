import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  return (
    <div className="bg-blue-50 p-8">
      <h2 className="text-3xl text-cyan-700 font-medium my-4">GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr className="my-4" />

      <ol>
        {categories.map((category) => (
          <GifGrid category={category} key={category} />
        ))}
      </ol>
    </div>
  );
};

export { GifExpertApp };
