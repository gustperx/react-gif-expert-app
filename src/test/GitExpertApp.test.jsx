import { shallow } from "enzyme";
import { describe, expect, test } from "vitest";
import { GifExpertApp } from "../GitExpertApp";

describe("Pruebas en el componente <GitExpertApp />", () => {
  test("debe de hacer match con el snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
