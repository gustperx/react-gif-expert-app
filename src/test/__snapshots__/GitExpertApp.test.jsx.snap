// Vitest Snapshot v1

exports[`Pruebas en el componente <GitExpertApp /> > debe de hacer match con el snapshot 1`] = `
<div
  className="bg-blue-50 p-8"
>
  <h2
    className="text-3xl text-cyan-700 font-medium my-4"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="my-4"
  />
  <ol />
</div>
`;

exports[`Pruebas en el componente <GitExpertApp /> > debe de mostrar una lista de categorias 1`] = `
<div
  className="bg-blue-50 p-8"
>
  <h2
    className="text-3xl text-cyan-700 font-medium my-4"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="my-4"
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
